aiida/engine/processes/functions.py

```diff
@@ -140,18 +141,30 @@ def run_get_node(*args, **kwargs):
 
             process = process_class(inputs=inputs, runner=runner)
 
-            def kill_process(_num, _frame):
-                """Send the kill signal to the process in the current scope."""
-                LOGGER.critical('runner received interrupt, killing process %s', process.pid)
-                process.kill(msg='Process was killed because the runner received an interrupt')
+            # Only add handlers for interrupt signal to kill the current process if we are not in a daemon runner
+            # Without this check, running process functions in a daemon worker would be killed if the daemon is shutdown
+            original_handler = None
+            original_signal = signal.SIGINT
```

Comment:
original_signal somehow suggests there will be a new signal as well, but there isn't right?
I guess you could call it `kill_signal`...

Reply:
same for the handler

Reply:
Good point for the signal, I will change it. However, I am not sure if the change for the handler would be good. This stems from the somewhat weird way of unbinding a signal in python. You cannot actually unbind it as such, you simply need to call `signal.signal(signal, handler)` again and here then `handler` needs to be what it was before. That is why I store that handle in the `original_handler` before overriding it, and then later you restore it